### ISSUE TITLE
wasmedge support in kubeedge

### ISSUE THREAD

**Issue by snstaberah** [kubeedge/kubeedge#4586]
**What happened and what you expected to happen**:
WasmEdge is a lightweight WebAssembly runtime that meets the OCI standard and is compatible with k8s. Compared with Linux containers, WasmEdge is a better alternative to edge lightweight containers. WasmEdge provides a capability-based security sandbox, makes few assumptions about the underlying operating system services, and supports multiple programming languages at the front end.

now a lot of edgecomputing project start to support  wasmedge as runtime in edgenode，is there a plan to test and integrate wasmedge in kubeedge？or we just need a document to explain how to use wasmedge in kubeedge？

for example
https://www.bookstack.cn/read/openyurt-0.6-zh/c87261040f968576.md

**Comment by Shelley-BaoYue** [comment]
Yes, we already have a plan for this work, and hope to synchronize our progress with the community asap. Thank you for your attention

**Comment by snstaberah** [comment]
thanks，waiting for it

**Comment by Shelley-BaoYue** [comment]
hi @snstaberah,  could you please provide me more information such as the actual scenarios for wasmedge support in Kubeedge?

**Comment by snstaberah** [comment]
> hi @snstaberah, could you please provide me more information such as the actual scenarios for wasmedge support in Kubeedge?

the main requirement  is linux container runtime such as docker/containerd  is not design for edge，they have many requirement for the os， but os in edgenode usually use minimize install and limited hardware，we think wasmedge is a better lightweight runtime sloution ， os-independent and provide better security；  

we are trying to use a binary to control cameras in edgenode，the binary is very sample，but containerd and docker image is a little heavy for our arm 32 RaspBerry Pi ；wasm image and wasmedge runtime maybe  better for this scenario，and  we are trying to use wasm in cloud nodes，too

**Comment by Shelley-BaoYue** [comment]
hi, @snstaberah, I have wrote an operation guide for running WasmEdge demo on KubeEdge. Welcome to review it and give your suggestions.  
https://github.com/kubeedge/community/pull/137

**Comment by snstaberah** [comment]
> hi, @snstaberah, I have wrote an operation guide for running WasmEdge demo on KubeEdge. Welcome to review it and give your suggestions. [kubeedge/community#137](https://github.com/kubeedge/community/pull/137)

thanks very much，i will check it this week

**Comment by Realexplorer** [comment]
hi,@Shelley-BaoYue, According to the demo in [](https://github.com/kubeedge/community/pull/137) ,I tried on two centos8 machine installation, encountered the following problem:
After the WebAssembly-based Image wasi-demo is deployed on the cloud node, `kubectl describe pod wasi-demo` shows that the deployment failed. The image was pulled from docker instead of containerd
The edge node did not find the container of wasi-demo by using `crictl ps -a` command, but found the container by using `docker ps -a`. Check the container log display "standard_init_linux.go:228: exec user process caused: exec format error"

The edge node environment is as follows：
```
containerd -v
containerd github.com/containerd/containerd v1.5.7 8686ededfc90076914c5238eb96c883ea093a8ba
wasmedge -v
wasmedge version 0.11.2
crun -v
crun version 1.8.1.0.0.0.16-7be9e
commit: 7be9ec2d03e31df0090c062c5bf6ebd241a7cd3b
rundir: /run/crun
spec: 1.0.0
+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +WASM:wasmedge +YAJL
```

**Comment by Shelley-BaoYue** [comment]
> 

hi, @Realexplorer , what is the kubeedge version you installed?  And you can execute `cat /etc/kubeedge/config/edgecore.yaml |grep "runtime"`  to see the runtime type config.

**Comment by Realexplorer** [comment]
kubeedge version:
```
NAME    STATUS   ROLES                           AGE        VERSION
master   Ready    control-plane,master     3d17h      v1.23.0
work      Ready    agent,edge                     43h          v1.23.15-kubeedge-v1.13.0
```

runtime type config:
```
[root@work kubeedge]# cat /etc/kubeedge/config/edgecore.yaml |grep "runtime"
      runtimeRequestTimeout: 2m0s
```

**Comment by Shelley-BaoYue** [comment]
Sorry, I got the wrong case. I need you to execute `cat /etc/kubeedge/config/edgecore.yaml |grep "Runtime"` again.
